src/index.ts

```diff
@@ -117,16 +121,36 @@ export class Pipeline implements Serializable {
       ? {
           explicitDependencies: true,
           cache: new Map(),
+          mutator: opts.mutator,
         }
       : {
           explicitDependencies: false,
+          mutator: opts.mutator,
         };
 
+    const steps = await this.toList(newOpts);
+    if (opts.mutator) {
+      for (const step of steps) {
+        if (step instanceof Step) {
```

Comment:
what else does `steps` have? Is this to filter out wait steps? If yes, can we:
```suggestion
        if (!(step instanceof Step)) {
          continue;
        }
```
to reduce nesting a bit?